src/cmds/tail.c

```diff
@@ -15,77 +15,82 @@
 
 #define SIZE 200
 
-static void read_last_n_lines(FILE *fp, int n) {
+static int read_last_n_lines(FILE *fp, int n) {
 	int lines_checked = 0;
-
-	unsigned long long position_of_last_line_to_read;
-	char buf_str[2*SIZE];
+	unsigned long long last_line_ptr;
+	char buf_str[SIZE];
 
 	if (fseek(fp, 0, SEEK_END)) {
 		printf("fseek() failed");
-		return ;
+		return -1;
 	}
-	position_of_last_line_to_read = ftell(fp);
+	last_line_ptr = ftell(fp);
 
-	while (position_of_last_line_to_read) {
-		if (!fseek(fp, --position_of_last_line_to_read, SEEK_SET)) {
+	while (last_line_ptr) {
+		if (!fseek(fp, --last_line_ptr, SEEK_SET)) {
 			if (fgetc(fp) == '\n') {
 				if (lines_checked++ == n) {
 					break;
 				}
 			}
 		} else {
 			perror("fseek() failed");
+			return -1;
 		}
 	}
 	if (--lines_checked != n) {
-		position_of_last_line_to_read = ftell(fp);
-		fseek(fp, --position_of_last_line_to_read, SEEK_SET);
+		last_line_ptr = ftell(fp);
+		fseek(fp, --last_line_ptr, SEEK_SET);
 	}
 	while (fgets(buf_str, sizeof(buf_str), fp)) {
 		printf("%s", buf_str);
 	}
+	return 0;
 }
 
 int main(int argc, char **argv) {
 	FILE *fp;
-	int opt, option_number = 0, lines = 10, k;
-	int digit_checker = 1;	
+	int opt, option_number = 0, lines = 10, k, temp;
+	int digit_checker = 1;
 
 	if (argc < 2) {
-		printf ("Usage: more [FILE]\n");
+		printf("Usage: more [FILE]\n");
 		return 0;
 	}
 
 	while (-1 != (opt = getopt(argc, argv, "n"))) {
 		option_number++;
 		switch (opt) {
 		case 'n':
-			for (k = 0; k < strlen(argv[option_number + 1]); k++) {
-				if (!isdigit(argv[option_number + 1][k])) digit_checker = 0;
+			temp = option_number + 1;
```

Comment:
You can declare temp in this block as I see. Also may be rename option_number to shorter name